src/app3.js

```diff
@@ -148,40 +239,80 @@ const https_options = {
 	cert: fs.readFileSync("cert.pem")
 };
 
-const server = https.createServer(https_options, (req, res) => {
-
-    let objResponse;
+let reqCount = 0;
+const server = https.createServer(https_options, async (req, res) => {
+    const currReqCount = reqCount;
+    console.log(logRequestPricingRecord(currReqCount), "requesting record...");
 
-    console.log(new Date().toUTCString() + " : " + req.connection.remoteAddress + " : requesting record...");
+	let version;
+	try {
+		const baseURL = 'https://' + req.headers.host + '/';
+		const reqUrl = new URL(req.url, baseURL);
+		version = Number(reqUrl.searchParams.get('version'));
+	} catch {
+		version = 0;
+	}
+	console.log(logRequestPricingRecord(currReqCount), "requesting version", version);
 
+    let response;
     const db = initDb(dbConfig);
     try {
-	db.query("SELECT *, UNIX_TIMESTAMP(Timestamp) AS UT FROM PricingRecord ORDER BY PricingRecordPK DESC LIMIT 1")
-	    .then(result => {
-		result[0].signature = result[0].Signature;
-		result[0].timestamp = result[0].UT;
-
-		// move btc to mv avg
-		result[0].xBTC = result[0].xBTCMA;
-		delete result[0].xBTCMA;
-
-		delete result[0].Signature;
-		delete result[0].Timestamp;
-		delete result[0].UT;
-		objResponse = {"pr":result[0]};
-		res.writeHead(200, "Content-Type: application/json");
-		res.write(JSON.stringify(objResponse));
-		console.log(JSON.stringify(objResponse));
+		const result = await db.query("SELECT * FROM PricingRecord ORDER BY PricingRecordPK DESC LIMIT 1");
+		console.log(logRequestPricingRecord(currReqCount), "record from DB: ", result);
+
+		// make sure record is recent enough - don't want to allow an old record through
+		// used INTERVAL + buffer to allow for latency variation in Chainlink + Coingecko + DB logic
+		const recordTimestamp = new Date(result[0].Timestamp);
+		if (new Date() - recordTimestamp > (INTERVAL + 10000)) {
+			console.log(logRequestPricingRecord(currReqCount), "Something is wrong, record is not recent enough.");
+			res.writeHead(404, "Content-Type: text/plain");
+			response = "Recent record not found";
+		} else {
+			// move btc to mv avg
+			result[0].xBTC = result[0].xBTCMA;
+			delete result[0].xBTCMA;
+
+			if (version < LATEST_VERSION) {
```

Comment:
Shouldn't this be a fixed point in space for all time? LATEST_VERSION will change in future in all likelihood...

Reply:
True, will update

Reply:
[Done](https://github.com/dweab/oracle-lite/commit/7eba1f3cfb6c4a9190c756d9e3ccf1e22931b5eb) :)